### DISCUSSION TITLE
Linux

### DISCUSSION THREAD

**Discussion by cchacons-tech** [terraform-docs/terraform-docs#637] [Q&A] [answered]
How do I install terraform-docs on ubuntu ?

**Comment by djkrivak** [comment]
Hi @cchacons-tech,

Refer to the [Pre-compiled Binary](https://terraform-docs.io/user-guide/installation/#pre-compiled-binary) section in the Installation section of docs.

**Comment by nileshgadgi** [comment, marked as answer]
Hello @cchacons-tech,
You can install terraform-doc using this script,
Just copy the below code and paste in your terminal.
```
curl -Lo ./terraform-docs.tar.gz https://github.com/terraform-docs/terraform-docs/releases/download/v0.16.0/terraform-docs-v0.16.0-$(uname)-amd64.tar.gz
tar -xzf terraform-docs.tar.gz
chmod +x terraform-docs
mv terraform-docs /usr/local/terraform-docs
```
You can install it via GoLang too if you have Go installed on your machine.

The latest version can be installed using go install or go get:
```
# go1.17+
go install github.com/terraform-docs/terraform-docs@v0.16.0
```
```
# go1.16
GO111MODULE="on" go get github.com/terraform-docs/terraform-docs@v0.16.0
```
NOTE: please use the latest Go to do this, minimum go1.16 is required.

This will put terraform-docs in $(go env GOPATH)/bin. If you encounter the error terraform-docs: command not found after installation then you may need to either add that directory to your $PATH as shown [here](https://golang.org/doc/code.html#GOPATH) or do a manual installation by cloning the repo and run make build from the repository which will put terraform-docs in:
```
$(go env GOPATH)/src/github.com/terraform-docs/terraform-docs/bin/$(uname | tr '[:upper:]' '[:lower:]')-amd64/terraform-docs
```

You can refer [this](https://github.com/terraform-docs/terraform-docs#installation) for more information.
